codegen/templates/endpoint.tmpl

```diff
@@ -87,6 +88,14 @@ func (handler *{{$handlerName}}) Register(g *zanzibar.Gateway) error {
 			{{- else -}}
 			handler.HandleRequest,
 			{{- end}}
+			g.AllHostScope.Tagged(map[string]string{
```

Comment:
Not a fan of creating and passing a subscope here, it's subscoped again at https://github.com/uber/zanzibar/pull/201/files#diff-7aed2fb9afc6ff1780de1ca94cd0a938R155. Why not pass down the tags info and create proper subscope once?

Reply:
It's subscoped with the `status-code` tag, which is only known when handling the request.  Instead, I'll need to pre-allocate scopes and counters for each status-code.